### PR TITLE
Checklist service

### PR DESCRIPTION
**Pull request by vaxinate**

### CODE REVIEW

File: app/models/checklister.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    'https://media.giphy.com/media/U2f3sQcYU40BG/giphy.gif',
+    'https://media.giphy.com/media/rzIwP3d5tBi8/giphy.gif',
+  ]
+
+  def initialize(payload)
+    self.payload = payload
+  end
+
+  # read the checklist file from the repo and return its text
+  def get_checklist_text
```
Comment (lintomat) [inline app/models/checklister.rb]: Method has too many lines. [13/10]

Comment (lintomat) [inline app/models/checklister.rb]: Do not prefix reader method names with `get_`.

File: app/models/checklister.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Add a release checklist from the repo any time commits are pushed to master
+# This doesn't do anything with slack, so the only required setup is repo webhooks
+
+class Checklister
```
Comment (lintomat) [inline app/models/checklister.rb]: Missing top-level class documentation comment.

File: app/models/checklister.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  end
+
+  def repo_name
+    payload['repository']['name']
+  end
+
+  def repo_owner_name
+    payload['repository']['owner']['name']
+  end
+
```
Comment (lintomat) [inline app/models/checklister.rb]: Extra empty line detected at class body end.

File: app/models/checklister.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# Add a release checklist from the repo any time commits are pushed to master
+# This doesn't do anything with slack, so the only required setup is repo webhooks
+
+class Checklister
+  attr_accessor :payload
+
+  CHECKLIST_PATH = 'RELEASE_CHECKLIST.md'
```
Comment (lintomat) [inline app/models/checklister.rb]: Freeze mutable objects assigned to constants.

File: app/models/checklister.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  DISAPPOINTMENT_GIFS = [
```
Comment (lintomat) [inline app/models/checklister.rb]: Freeze mutable objects assigned to constants.

File: app/models/checklister.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    HEREDOC
+  end
+
+  def maybe_checklist
+    return unless needs_checklist?
+
+    github.issues.create(
+      repo_owner_name,
+      repo_name,
+      title: "Release Checklist -- #{Date.today}",
```
Comment (lintomat) [inline app/models/checklister.rb]: Do not use `Date.today` without zone. Use `Time.zone.today` instead.

**Comment by lintomat** [comment]
No tests added or edited in this PR. Pull Requests should have tests. 1 of 1 this week have tests.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.herokuapp.com/relint/revelrylabs/slack-agile/105).

**Comment by lintomat** [comment]
No tests added or edited in this PR. Pull Requests should have tests. 0 of 1 this week have tests.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.herokuapp.com/relint/revelrylabs/slack-agile/105).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.herokuapp.com/relint/revelrylabs/slack-agile/105).

File: app/models/checklister.rb (hunk 1 above)
Comment (vaxinate) [inline app/models/checklister.rb]: this isn't just a reader, it's actually getting something from the web.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.herokuapp.com/relint/revelrylabs/slack-agile/105).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.herokuapp.com/relint/revelrylabs/slack-agile/105).

**Pull request merged**